File: backend/app/models/step_record.py
from sqlalchemy import Column, Integer, Float, Date, DateTime, ForeignKey, String 
from sqlalchemy.sql import func
from app.database import Base

class StepRecord(Base):
    __tablename__ = "step_records"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    date = Column(Date, nullable=False)
    steps = Column(Integer, nullable=False)
    distance_km = Column(Float, nullable=True)
    calories_burned = Column(Float, nullable=True)
    source = Column(String, default="healthkit")
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: frontend/types/StepRecord.ts
export interface StepRecord {
  id: number;
  user_id: number;
  date: string;
  steps: number;
  distance_km: number | null;
  calories_burned: number | null;
  source: string;
}

export interface StepRecordCreate {
  date: string;
  steps: number;
  distance_km?: number | null;
  calories_burned?: number | null;
  source?: string;
}